test/unit/BuilderTest.php

```diff
@@ -567,12 +567,27 @@ public function withClaimMustKeepAFluentInterface()
      * @covers ::__construct
      * @covers ::withClaim
      * @covers \Lcobucci\JWT\Token\RegisteredClaimGiven
+     *
+     * @dataProvider dataWithClaimDeprecationNotice
      */
-    public function withClaimShouldThrowExceptionWhenTryingToConfigureARegisteredClaim()
+    public function withClaimShouldSendDeprecationNoticeWhenTryingToConfigureARegisteredClaim(string $claim, $value)
     {
-        $this->expectException(RegisteredClaimGiven::class);
+        $this->expectDeprecation('The use of the method "withClaim" is deprecated for registered claims. Please use dedicated method instead.');
+
+        $this->createBuilder()->withClaim($claim, $value);
+    }
 
-        $this->createBuilder()->withClaim('sub', 'me');
+    public function dataWithClaimDeprecationNotice()
+    {
+        return [
+            ['sub', 'me'],
+            ['aud', 'him'],
+            ['jti', '0123456789ABCDEF'],
+            ['iss', 'you'],
+            ['exp', time()+3600],
+            ['iat', time()],
+            ['nbf', time()],
+        ];
```

Comment:
We should also have the expected values here (and use `self::assertEquals($expected, $token->claims()->get($name))` on the test):

```suggestion
        $now = time();
        $nowAsDate = new DateTimeImmutable('@' . $now);

        return [
            ['sub', 'me', 'me'],
            ['aud', 'him', ['him']],
            ['jti', '0123456789ABCDEF', '0123456789ABCDEF'],
            ['iss', 'you', 'you'],
            ['exp', $now+3600, $nowAsDate->modify('+1 hour')],
            ['iat', $now, $nowAsDate],
            ['nbf', $now, $nowAsDate],
        ];
```

Reply:
Thank you for the review.
The tests have been updated accordingly.